File: Python3/프로그래머스/1/12944. 평균 구하기/평균 구하기.py
def solution(arr):
    answer=0
    for i in range(len(arr)):
        answer= (answer*i + arr[i])/(i+1)
    return answer

def main():
    arr1 = [1, 2, 3, 4]
    print(f"average: {solution(arr1)}")

    
    arr2 = [5, 5]
    print(f"average: {solution(arr2)}")

if __name__ == "__main__":
    main()

File: C++/프로그래머스/1/12944. 평균 구하기/평균 구하기.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

double solution(vector<int> arr) {
    double answer = 0;
    for (int i = 0; i < arr.size(); i++) {
        answer = (answer * (i) + arr[i]) / (i+1);

    }

    return answer;
}

int main(){
    vector<int> arr1 = {1,2,3,4};
    cout << "average: " << solution(arr1) << endl;

    vector<int> arr2 = {5, 5};
    cout << "average: " << solution(arr2) << endl;

    return 0;

}